pkg/utils/cniutils/cni_utils.go

```diff
@@ -74,6 +75,22 @@ func WaitForAddressesToBeStable(netLink netlinkwrapper.NetLink, ifName string, t
 	}
 }
 
+// GetNodeMetadata calling node local imds metadata service using provided key
+// return either a non-empty value or an error
+func GetNodeMetadata(key string) (string, error) {
```

Comment:
Why do we need another function here? Seems like callers can just call `imds.GetMetaData(key)` directly.

Reply:
Whole lopic inside GetNodeMetadata has loop to deal with cases like no error but value still empty.
```
	for {
		value, err = imds.GetMetaData(key)
		if err != nil {
			return "", err
		}
		if value != "" {
			return value, nil
		}
	}
```

Reply:
Is the concern that the ipv6 address may not be available in IMDS yet, so we get an empty value and retry until it is? I think we can skip that, as if the address is not available, then aws-node pod will error and restart, at which case it will become available. My concern with this for loop is that it could run indefinitely if no IPv6 addr is configured.

Reply:
if no ipv6 configured, it will return err which is some kind of 404, so it will not run indefinitely.
I am not the original author who added the for, I am guessing it is for imds service throttling.

Reply:
I see, we can leave it as is then